Sources/BSON/BSONRegularExpression.swift

```diff
@@ -64,6 +64,78 @@ public struct BSONRegularExpression: Equatable, Hashable {
 }
 
 extension BSONRegularExpression: BSONValue {
+    /*
+     * Initializes a `BSONRegularExpression` from ExtendedJSON.
+     *
+     * Parameters:
+     *   - `json`: a `JSON` representing the canonical or relaxed form of ExtendedJSON for a `RegularExpression`.
+     *   - `keyPath`: an array of `String`s containing the enclosing JSON keys of the current json being passed in.
+     *              This is used for error messages.
+     *
+     * Returns:
+     *   - `nil` if the provided value is not a `RegularExpression` with valid options.
+     *
+     * Throws:
+     *   - `DecodingError` if `json` is a partial match or is malformed.
+     */
+    internal init?(fromExtJSON json: JSON, keyPath: [String]) throws {
+        switch json {
+        case let .object(obj):
+            // canonical and relaxed extended JSON
+            guard let value = obj["$regularExpression"] else {
+                return nil
+            }
+            guard obj.count == 1 else {
+                throw DecodingError._extendedJSONError(
+                    keyPath: keyPath,
+                    debugDescription: "Expected only \"$regularExpression\" key, found too many keys: \(obj.keys)"
+                )
+            }
+            guard let regexArr = value.arrayValue else {
```

Comment:
the regular expression format is in the format `{"pattern": <string>, "options": <string> }`, not an array.

Reply:
I was going back and forth on this, I think the documentation is missing the quotes around "pattern" which confused me. Ill fix this